docs/architecture/limits.md

```diff
@@ -0,0 +1,42 @@
+# Limits
+
+The following are the current limits enforced in the `miden-base` and `miden-node`:
+
+## Accounts
+- Max assets per account: **no limit**.
+- Max top-level storage slots per account: **255**. Each storage slot can contain an unlimited 
+  amount of data (e.g., if the storage slot contains an array or a map).
+- Max code size per account: **no limit** (but we plan to enforce code size limit in the future, 
+  at least for public accounts).
+
+## Notes
+- Min assets per note: **1**.
+- Max assets per note: **256**.
+- Max inputs per note: **128**. The value can be represented using as a single byte while being 
+  evenly divisible by 8.
+- Max code size per note: **no limit** (but we plan to enforce code size limit in the future,
+  at least for public notes).
+
+## Transactions
+- Max input notes per transaction: **1024**.
+- Max output notes per transaction: **1024**.
+- Max code size of tx script: **no limit** (but we plan to enforce code size limit in the future,
+  at least for public notes).
+
+## VM Cycles
+- Max number of VM cycles: technically **no limit**, but practically a single transaction cannot take 
+  more than $2^{29}$ cycles. Even more practically, anything above $2^{20}$ cycles will require 10GB+ 
+  of RAM (e.g., $2^{24}$ cycles will require ~256GB of RAM, though, with optimizations, we can bring 
+  it down to ~128GB).
```

Comment:
Let's set minimum here at $2^{10}$ cycles (and we can also enforce this). As for maximum, I'm not sure we need such an elaborate explanation here. Also, let's set it to $2^{30}$.

Reply:
It's unclear for me what does minimum VM cycles mean. Should we discard transactions which took less than **$2^{10}$** VM cycles during execution? Is it needed for safety against attacks?

Reply:
We could enforce this, but won't have too much of a practical consequence because transaction kernel currently takes around 3K cycles to execute. So, all transactions will likely take over 3K - 4K cycles regardless of the limit.